### DISCUSSION TITLE
Building .zip sdist files on posix systems

### DISCUSSION THREAD

**Discussion by matthewsht** [pypa/setuptools#2982] [Q&A] [answered]
All,
   I'm sorry to ask this here, but google (quite a lot of google) has failed me. I'm not even certain this is setuptools or build question.

I have multiple python packages for talking to our internal systems. These are system independant - however, I'm on a mac and want to build .zip based sdists so my colleagues on windows can easily pick up development.

How do I tell my setup.cfg to only (or at least also) build .zip sdists?

**Comment by abravalheri** [comment]
Hi @matthewsht, have you tried to add the following section to your setup.cfg?


```ini
[sdist]
formats = zip
```

However, this will not work with modern ways of creating distributions.

There is a spec (PEP 517) that requires sdists to be tar.gz files...

**Reply by matthewsht** [reply]
I just added  that to my setup.cfg and the build still built a tar.gz sdists.

Relevent bits from the build include:
Writing npmp-0.8/setup.cfg
Creating tar archive
....
Successfully built npmp-0.8.tar.gz and npmp-0.8-py3-none-any.whl

I'm on  setuptools         59.6.0  due to being on python 3.6
My build version is 0.7.0

**Reply by abravalheri** [reply]
Are you using the `build` package from PyPI (e.g. by running `python -m build`)?

If that is the case, I am afraid tar.gz is the only thing you can produce... (The `build` package follows the Python spec I mentioned before, PEP 517).

**Reply by abravalheri** [reply]
(Your colleagues using Windows should still be able to install tar.gz files with pip, and if they are willing to use the command line, they can still have access to the `tar` command on the CLI, which is present on Windows. Otherwise they can always use something like the 7zip program)

**Reply by matthewsht** [reply]
I am using PyPI build (0.7.0). I think thats a shortsight in PEP 517, but thats clearly not a setuptools problem.

Thank You! for the fast response and explanation - its appreciated.

**Comment by matthewsht** [comment, marked as answer]
From abravalheri above, the answer to my question is "you can't" and probably (to be PEP 517 compatible) "you don't want to".
